### PR TITLE
Ruby group capstone - Catalog of my things

### LINKED ISSUE #1: [0.30pt] Group-tasks
Create Item class in a separate .rb file.
- All Item class properties visible in the diagram should be defined and set up in the constructor method. _Exception: properties for the 1-to-many relationships should NOT be set in the constructor method. Instead, they should have a custom setter method created._

### LINKED ISSUE #4: [1pt] Team-member-1-Tasks-3
- Implement methods:
    - add_item method in the Label class
        - should take an instance of the Item class as an input
        - should add the input item to the collection of items
        - should add self as a property of the item object (by using the correct setter from the item object)
    - can_be_archived?() in the Book class
        - should override the method from the parent class
        - should return true if parent's method returns true OR if cover_state equals to "bad"
        - otherwise, it should return false

### LINKED ISSUE #5: Ruby group capstone: set up your Kanban board
In this pull request, we have

- Created an empty repository for our group project.
- Followed the steps described in [curriculum-tools-copy-projects](https://github.com/microverseinc/curriculum-tools-copy-projects) in order to copy the [[TEMPLATE] Ruby Group Capstone Project](https://github.com/microverseinc/curriculum-ruby/projects/1) board and opened the GitHub project (Kanban board) in our team repo.
- Created tasks on our Kanban board based on the [project requirements divided by team member](https://github.com/microverseinc/curriculum-ruby/blob/main/group-capstone/catalog_of_my_things_divided.md).
- Each task has a separate card, descriptive title, points in the [Xpt] prefix, and a description of the task.
- Each of the team members has been assigned to the cards as follows:
- 
 🟢 Group-Tasks: @Paul-js-hub @udeaghad @CrystallineButterfly 
🟢 Team-Member-1: @udeaghad 
🟢 Team-Member-2: @Paul-js-hub 
🟢 Team-Member-3: @CrystallineButterfly 

Full Kanban Board link is https://github.com/Paul-js-hub/my-catalog/projects/1 ☑️

Kindly review and approve

### LINKED ISSUE #6: [0.30pt] Group Task
- Add all methods visible in the diagram.
- Implement methods:
    - can_be_archived?() in the Item class
        - should return true if published_date is older than 10 years
        - otherwise, it should return false
    - move_to_archive() in the Item class
        - should reuse can_be_archived?() method
        - should change the archived property to true if the result of the can_be_archived?() method is true
        - should do nothing if the result of the can_be_archived?() metho

### LINKED ISSUE #12: [1pt] Team Member 2 task 3
- Add unit tests for all implemented methods.
 - The following options should be available:
    - List all music albums
    - List all genres (e.g 'Comedy', 'Thriller')
    - Add a music album

### LINKED ISSUE #13: [1pt] Team Member 2 task 4
- All data should be preserved by saving collections in .json files.
 - Create a schema.sql file with tables that will be analogical to the structure of the classes that you created:
    - music_albums table (add all properties and associations from the parent Item class as table columns)
    - genres table

### LINKED ISSUE #14: [1pt] Team Member 3 task 1
- Create a Game class in a separate .rb file.
- Create an Author class with an association to the Item class (in a separate .rb file).
- All Game class properties visible in the diagram should be defined and set up in the constructor method.

### LINKED ISSUE #15: [1pt] Team Member 3 task 2
- All Author class properties visible in the diagram should be defined and set up in the constructor method.
- Implement methods:
    - add_item method in the Author class
        - should take an instance of the Item class as an input
        - should add the input item to the collection of items
        - should add self as a property of the item object (by using the correct setter from the item object)
    - can_be_archived?() in the Game class
        - should override the method from the parent class
        - should return true if parent's method returns true AND if last_played_at is older than 2 years
        - otherwise, it should return false

### LINKED ISSUE #16: [1pt] Team Member 3 task 3
- Add unit tests for all implemented methods.
- The following options should be available:
    - List of games
    - List all authors (e.g. 'Stephen King')
    - Add a game

### LINKED ISSUE #17: [1pt] Team Member 3 task 4
- All data should be preserved by saving collections in .json files.
 - Create a schema.sql file with tables that will be analogical to the structure of the classes that you created:
    - games table (add all properties and associations from the parent Item class as table columns)
    - authors table

### PR DESCRIPTION
**Pull request by CrystallineButterfly**
Hi, Thank you for reviewing this PR 🤖

⚫️ Insert and query data in SQL ✅
⚫️ Use primary key & foreign key mechanisms for joining tables ✅
⚫️ Understand the different types of relationships between tables ✅
⚫️ Query multiple tables ✅
⚫️ Prepare complex queries that answer analytical questions ✅
⚫️ Run a program using the command line ✅
⚫️ Use Ruby syntax for basic programming operations ✅
⚫️ Apply Ruby best practices and language style guides in code ✅
⚫️ Describe the SOLID principles of OOP ✅
⚫️ Implement classes and objects in Ruby ✅
⚫️ Understand the four main principles of OOP ✅
⚫️ Implement encapsulation and inheritance with Ruby ✅
⚫️ Create a UML class diagram ✅
⚫️ Explain the difference between associations, aggregations, and composition in OOP ✅
⚫️ Set up associations between classes and objects ✅
⚫️ Store data in files ✅
⚫️ Build interactive console apps ✅
⚫️ Recognize the value of making equal contributions to group projects to produce the best outcome ✅

🤖 This was a group project made by @udeaghad @Paul-js-hub @CrystallineButterfly 🤖

Here is demo vid = https://drive.google.com/file/d/1-_ZcX_txvMtpUxhLJvl4P9mi5Eu1VEA5/view?usp=sharing

All the best- K, D, P

### CODE REVIEW

File: schema/schema.sql [hunk 1]
```diff
@@ -0,0 +1,48 @@
+CREATE TABLE author (
```
Comment (gemmen29) [inline schema/schema.sql]: - [x] Kindly make sure that the schema file is working fine as now when I get an error when I try to run the code.

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by CrystallineButterfly** [comment]
Hello there, 

Thank you for the suggestions! 

What error were you getting? @gemmen29

File: schema/schema.sql (hunk 1 above)
Comment (CrystallineButterfly) [reply schema/schema.sql]: Solved!

**Review by KDlamini — APPROVED** [review]
### Hi team @CrystallineButterfly, @Paul-js-hub, @udeaghad 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the required changes. 🎊 🥳 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!

## HIGHLIGHTS 🌟 

✔️ Good PR title and description
✔️ Basic UI implemented and works as expected
✔️ Data is preserved using JSON
✔️ Commit messages are meaningful
✔️ No linter errors

# Status: Approved ✔️ 

oscars-standing-ovation
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by CrystallineButterfly** [comment]
Hi @KDlamini Thank you! Merged now! Have a good rest of your day!

**Pull request merged**